Sources/CovidCertificateSDK/TrustList/TrustListManager.swift

```diff
@@ -0,0 +1,131 @@
+
+    var trustStorage: TrustStorageProtocol { get }
+
+    func restartTrustListUpdate(completionHandler: @escaping (() -> Void), updateTimeInterval: TimeInterval)
+}
+
+class TrustlistManager: TrustlistManagerProtocol {
+    // MARK: - Components
+
+    var trustStorage: TrustStorageProtocol
+    var revocationListUpdater: TrustListUpdate
+    var trustCertificateUpdater: TrustListUpdate
+    var nationalRulesListUpdater: TrustListUpdate
+
+    private let operationQueue = OperationQueue()
+    private var timer: Timer?
+
+    // MARK: - Init
+
+    init() {
+        trustStorage = TrustStorage()
+        nationalRulesListUpdater = NationalRulesListUpdate(trustStorage: trustStorage)
+        revocationListUpdater = RevocationListUpdate(trustStorage: trustStorage)
+        trustCertificateUpdater = TrustCertificatesUpdate(trustStorage: trustStorage)
+    }
+
+    func restartTrustListUpdate(completionHandler: @escaping (() -> Void), updateTimeInterval: TimeInterval) {
+        timer = Timer.scheduledTimer(withTimeInterval: updateTimeInterval, repeats: true, block: { [weak self] _ in
+            guard let strongSelf = self else { return }
+            strongSelf.forceUpdate(completionHandler: completionHandler)
+        })
+
+        timer?.fire()
+    }
+
+    private func forceUpdate(completionHandler: @escaping (() -> Void)) {
+        let group = DispatchGroup()
+
+        for updater in [revocationListUpdater, trustCertificateUpdater] {
+            group.enter()
+
+            updater.addCheckOperation(checkOperation: { _ in
+                group.leave()
+            }, forceUpdate: true)
+        }
+
+        group.notify(queue: .main) {
+            completionHandler()
+        }
+    }
+}
+
+public class TrustListUpdate {
+    // MARK: - Operation queue handling
+
+    private let operationQueue = OperationQueue()
+
+    private var updateOperation: Operation?
+
+    internal var lastUpdate: Date?
+    private var lastError: NetworkError?
+
+    internal let trustStorage: TrustStorageProtocol
+
+    // MARK: - Add Check Operation
+
+    init(trustStorage: TrustStorageProtocol) {
+        // ensures that the update request is done before the other tasks
+        self.trustStorage = trustStorage
+        operationQueue.maxConcurrentOperationCount = 1
+    }
+
+    public func addCheckOperation(checkOperation: @escaping ((NetworkError?) -> Void), forceUpdate: Bool = false) {
+        let updateNeeeded = !isListStillValid() || forceUpdate
+        let updateAlreadyRunnning = updateOperation != nil
+
+        if updateNeeeded, !updateAlreadyRunnning {
+            updateOperation = BlockOperation(block: { [weak self] in
+                guard let strongSelf = self else { return }
+                strongSelf.startUpdate()
+            })
+
+            // !: initialized just above
+            operationQueue.addOperation(updateOperation!)
+        }
+
+        operationQueue.addOperation {
```

Comment:
Doesn't this mean, that if the app is opened (which `restartTrustListUpdate`) while the available data is not yet expired, the force-update operation will block (up to 30s) verifying operations. Even though the data is not yet expired and we're just pro-actively trying to download an updated list.